src/miner.cpp

```diff
@@ -151,6 +155,7 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
     coinbaseTx.vout.resize(1);
     coinbaseTx.vout[0].scriptPubKey = scriptPubKeyIn;
     coinbaseTx.vout[0].nValue = nFees + GetBlockSubsidy(nHeight);
+    StakingPool::getInstance()->increaseBalance(GetStakingRewardForHeight(nHeight));
```

Comment:
Staking pool should be modified only if a new, already mined block appears. We shouldn't increase it when assembling a new block.